src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] hashTable;
+    private int threshold = DEFAULT_INITIAL_CAPACITY * (int) DEFAULT_LOAD_FACTOR;
+    private int size;
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold || hashTable == null) {
```

Comment:
instead of this null check just add a constructor to initialize all non-constant values like our table and threshold

Reply:
Yes, I originally wanted to initialize the HashMap in the constructor, but I decided to leave this check by analogy with the original implementation.